src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+                return fareType.equals(Fare.FareType.orcaSenior) ? 1.00f : defaultFare;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 3.00f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 2.50f;
+            case KC_METRO:
+            case SOUND_TRANSIT:
+                return 1.00f;
+            default: return defaultFare;
+        }
+    }
+
+    /**
+     * Apply Orca youth discount fares based on the ride type.
+     */
+    private float getYouthFare(RideType rideType, float defaultFare) {
+        switch (rideType) {
+            case COMM_TRANS_LOCAL_SWIFT: return 1.75f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 3.00f;
+            case PIERCE_COUNTY_TRANSIT: return 1.00f;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
+            case KITSAP_TRANSIT: return 2.00f;
+            case KC_METRO:
+            case SOUND_TRANSIT:
+            case EVERETT_TRANSIT:
+            case SEATTLE_STREET_CAR:
+                return 1.50f;
+            default: return defaultFare;
+        }
+    }
+
+    /**
+     * Get the ride price for a single leg. If testing, this class is being called directly so the required agency cash
+     * values are not available therefore the default test price is used instead.
+     */
+    private float getRidePrice(Ride ride, Fare.FareType fareType, Collection<FareRuleSet> fareRules) {
+        if (IS_TEST) {
+            // Testing, return default test ride price.
+            return DEFAULT_TEST_RIDE_PRICE;
+        }
+        List<Ride> ridesSingleton = new ArrayList<>();
+        ridesSingleton.add(ride);
+        return calculateCost(fareType, ridesSingleton, fareRules);
+    }
+
+    /**
+     * Calculate the cost of a journey. Where free transfers are not permitted the cash price is use. If free transfers
+     * are applicable, the most expensive discount fare across all legs is added to the final cumulative cash price.
+     */
+    @Override
+    public boolean populateFare(Fare fare,
+                                   Currency currency,
+                                   Fare.FareType fareType,
+                                   List<Ride> rides,
+                                   Collection<FareRuleSet> fareRules
+    ) {
+        float cost = 0;
+        float orcaFareDiscount = 0;
+        for (Ride ride : rides) {
+            RideType rideType = classify(ride.routeData);
+            float singleLegPrice = getRidePrice(ride, fareType, fareRules);
+            float discountedFare = getDiscountedFare(fareType, rideType, singleLegPrice);
+            if (hasFreeTransfers(fareType, rideType)) {
+                // If using Orca (free transfers), the total fare should be equivalent to the
+                // most expensive leg of the journey.
+                orcaFareDiscount = Float.max(orcaFareDiscount, discountedFare);
+            } else {
```

Comment:
The calculation of Free Transfers in practice also has a time limit of 2 hours from the time of the boarding of the first leg of a journey ([see here](https://www.soundtransit.org/ride-with-us/how-to-pay/orca-card)). Although rare, there are situations where a single journey ([this one for example](https://newtripplanner.communitytransit.org/#/?ui_activeSearch=2cr9ajfcm&ui_activeItinerary=0&fromPlace=930%20Chestnut%20Street%2C%20Bremerton%2C%20WA%2C%20USA%3A%3A47.58717856130287%2C-122.62266585140499&toPlace=221%208th%20Avenue%20North%2C%20Edmonds%2C%20WA%2C%20USA%3A%3A47.811771147718936%2C-122.36988575267044&date=2021-07-14&time=11%3A20&arriveBy=false&mode=WALK%2CBUS%2CTRAM%2CFERRY%2CRAIL&showIntermediateStops=true&maxWalkDistance=1207&optimize=QUICK&walkSpeed=1.34&ignoreRealtimeUpdates=true&companies=&bannedRoutes=&numItineraries=3&otherThanPreferredRoutesPenalty=900&preferredRoutes=)) can have a leg that begins 2 hours after the first one, thus requiring the purchase of a new fare.

Reply:
@evansiroky I've made an update. I've also made a few assumptions! Will need further review.